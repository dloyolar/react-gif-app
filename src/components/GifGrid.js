import React, { useState, useEffect } from 'react';

export const GifGrid = ({ category }) => {
  const [count, setCount] = useState(0);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      'https://api.giphy.com/v1/gifs/search?q=one+piece&limit=25&api_key=YzfEcHSohd23s7SuJ3HGLmsMEzUGtXY0';
    const res = await fetch(url);
    const { data } = await res.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    console.log(gifs);
  };

  return (
    <>
      <h3> {category} </h3>
      <h3>{count}</h3>
      <button onClick={() => setCount(count + 1)}></button>
    </>
  );
};
