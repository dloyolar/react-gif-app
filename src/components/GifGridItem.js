import React from 'react';

export const GifGridItem = ({ id, title, url }) => {
  return (
    <div className='card animate__animated animate__fadeInDown'>
      <img src={url} alt={title} loading='lazy' />
      <p>{title.toUpperCase() ? title : 'Imagen without title ğŸ˜¢'}</p>
    </div>
  );
};
