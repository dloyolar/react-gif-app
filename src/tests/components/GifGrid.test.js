import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('test <GifGrid />', () => {
  const category = 'Haikyuu';
  test('should mount <GifGrid />', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should be show items when images is loaded useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/cualquier/imagen.jpg',
        title: 'Any Image',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
