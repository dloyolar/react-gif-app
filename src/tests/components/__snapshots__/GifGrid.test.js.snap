// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifGrid /> should mount <GifGrid /> 1`] = `
<Fragment>
  <h3>
     
    Haikyuu
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
